src/analyses/goto_rw.cpp

```diff
@@ -145,7 +145,8 @@ void rw_range_sett::get_objects_byte_extract(
   else
   {
     index*=8;
-    if(index>=pointer_offset_bits(be.op().type(), ns))
+    if(index<0 ||
+       index>=pointer_offset_bits(be.op().type(), ns))
```

Comment:
I'm afraid this code has more flaws (yes, my fault): the offset must be computed relative to the root object, which may or may not be the same as `be.op()`. Consider the case of a byte_extract within a struct member (and the test that's now supposedly fixed is exactly of that nature).